apps/backend/src/db/migrations/20260719120000_workspace_integrations_multi_install.sql

```
@@ -0,0 +1,35 @@
+-- install mid-deploy (a retryable 500). Token refreshes are UPDATEs and are
+-- unaffected.
+
+CREATE UNIQUE INDEX IF NOT EXISTS workspace_integrations_workspace_provider_installation
+    ON workspace_integrations (workspace_id, provider, installation_id)
+    WHERE installation_id IS NOT NULL;
+
+CREATE UNIQUE INDEX IF NOT EXISTS workspace_integrations_workspace_provider_no_installation
+    ON workspace_integrations (workspace_id, provider)
+    WHERE installation_id IS NULL;
+
+DROP INDEX IF EXISTS workspace_integrations_workspace_provider;
```

SQLFluff (4.2.2)

[error] DROP statement should use CONCURRENTLY to avoid locking the table.

(PG01)